fe/lib/components/SalaryDetails/SalaryDetails.tsx

```diff
@@ -0,0 +1,165 @@
+          const type = value as SalaryType;
+
+          setSalaryType(type);
+          onBlur({
+            key: 'basisCode',
+            salary: {
+              code: type,
+              interval: type === 'Hourly' ? 'Hour' : 'Year',
+            },
+          });
+        }}
+        tone={salaryTypeValidation.tone}
+        message={salaryTypeValidation.message}
+        label="Pay type"
+      >
+        {salaryTypes.map(({ label, value }) => (
+          <RadioItem key={value} label={label} value={value} />
+        ))}
+      </RadioGroup>
+
+      <Stack space="small">
+        <Text weight="strong">Pay range {currencies[currency]}</Text>
+        <Text size="small" tone="secondary">
+          Select a pay range to offer candidates.
+        </Text>
+        <Columns space="medium" collapseBelow="tablet">
+          <Column width="1/2">
+            <TextField
+              id="minimumAmount"
+              aria-label="minimum-amount"
+              onChange={({ currentTarget: { value } }) => setMinAmount(value)}
+              onBlur={({ currentTarget: { value } }) =>
+                onBlur({ key: 'minimumAmount', amount: value })
+              }
+              onClear={() => onBlur({ key: 'minimumAmount', amount: '' })}
+              value={minAmount}
+              tone={minAmountValidation.tone}
+              message={minAmountValidation.message}
+              placeholder="Minimum"
+              type="number"
+            />
+          </Column>
+          <Column width="1/2">
+            <TextField
+              id="maximumAmount"
+              aria-label="maximum-amount"
+              onChange={({ currentTarget: { value } }) => {
+                setMaxPay({ maxAmount: value, blurredMaxAmount });
+              }}
+              onBlur={({ currentTarget: { value } }) => {
+                setMaxPay({ maxAmount, blurredMaxAmount: value });
+                onBlur({ key: 'maximumAmount', amount: value });
+              }}
+              onClear={() => onBlur({ key: 'maximumAmount', amount: '' })}
+              value={maxAmount}
+              tone={maxAmountValidation.tone}
+              message={maxAmountValidation.message}
+              placeholder="Maximum"
+              type="number"
+            />
+          </Column>
+        </Columns>
+      </Stack>
+
+      <TextField
+        id="salaryDescription"
+        label="Pay shown on your ad"
+        secondaryLabel="Optional"
+        onChange={({ currentTarget }) => setDescription(currentTarget.value)}
+        onBlur={({ currentTarget: { value } }) =>
+          onBlur({ key: 'description', description: value })
+        }
+        onClear={() => onBlur({ key: 'description', description: '' })}
+        value={description}
+        placeholder={'Example content'}
+        tone={descriptionValidation.tone}
+        message={
+          descriptionValidation.message ?? 'e.g $50,000 + car + annual bonus'
+        }
+        characterLimit={MAX_CHAR_LIMIT}
```

Comment:
just to clarify, I don't see limit in the [description](https://developer.seek.com/schema/#/named-type/RemunerationPackage/field/descriptions)

Reply:
Oh yea that's a good point, I must've mistaken this for the hirer job reference 😅 , thanks for catching that, I'll remove that validation

Reply:
It looks like it *is* limited to 50 bytes:
<https://github.com/SEEK-Jobs/ie-position-posting-api/blob/2de31c458730e59a58cded1343fd473f32572857/src/http/controllers/helpers/validators/validateRemunerationDescriptions.ts#L1>

And the limit is documented on the input type:
<https://developer.seek.com/schema/#/named-type/RemunerationPackageInput/field/descriptions>

We usually don't document the limit on the output types because we don't enforce limits on the data we return, only what we accept.

Reply:
Oh man, if only I had a waited another minute

Reply:
ohh, sorry for confusing you by the wrong reference in schema :(